### DISCUSSION TITLE
Standard MQTT Data Link

### DISCUSSION THREAD

**Discussion by smcHannah** [yamcs/yamcs#916] [Q&A]
Obviously YAMCS already has standard UDP and TCP data links, but with more ground station networks using MQTT is there any standard MQTT configuration for YAMCS available?

**Comment by xpromache** [comment]
Not yet but there might be soon.
Do you need it for packets or CCSDS frames (unfortunately we have to make these separate in Yamcs as it is now)?

**Comment by xpromache** [comment]
I pushed some initial code for MQTT support:
https://github.com/yamcs/yamcs-mqtt

It depends on Yamcs 5.10 which will be released soon.
In the meanwhile you can test it by compiling Yamcs 5.10.0-SNAPSHOT manually.  Or you can have a look to see if it does what you expect.

**Comment by xpromache** [comment]
I published an update allowing to use your own converter between TM/TC frames/packets and MQTT messages. I found out that LeafSpace uses a json format for the TM packets/frames so I added a converter for that.

If you know of formats used by other ground stations I would be interested to find out.
